esyi/Solver.ml

```diff
@@ -362,26 +362,9 @@ let solveDependencies ~installed ~strategy dependencies solver =
       % p filenameOut
     ) in
 
-    let f p =
-      let readAndClose ic =
-        Lwt.finalize
-          (fun () -> Lwt_io.read ic)
-          (fun () -> Lwt_io.close ic)
-      in
-      let%lwt stdout = readAndClose p#stdout
-          and stderr = readAndClose p#stderr in
-      match%lwt p#status with
-      | Unix.WEXITED 0 ->
-        RunAsync.return ()
-      | _ ->
-        RunAsync.errorf
-          "@[<v>command failed: %a@\nstderr:@[<v 2>@\n%a@]@\nstdout:@[<v 2>@\n%a@]@]"
-          Cmd.pp cmd Fmt.lines stderr Fmt.lines stdout
-    in
-
-    let cmd = Cmd.getToolAndLine cmd in
     try%lwt
-      Lwt_process.with_process_full cmd f
+      let%bind env = EsyLib.EsyBashLwt.getMingwEnvironmentOverride () in
+      ChildProcess.run ~env cmd
```

Comment:
@andreypopp - are you OK with the refactoring here? It means there is a slight change in the error messages, but we still do get error messages from the process (just using `ChildProcess.run`'s faculty for it).

Reply:
Looks ok! Thank you!